pkg/didmethod/peer/generate.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package peer
+
+import (
+	"crypto"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+)
+
+const numalgo = "1"
+const encalgo = "1"
+
+//GeneratePeerDid creates the peer DID. For example : did:peer:11-479cbc07c3f991725836a3aa2a581ca2029198aa420b9d99bc0e131d9f3e2cbe
+func GeneratePeerDid(doc *did.Doc) (string, error) {
```

Comment:
I feel the function should take the publicKey, authentication details as the input param. ie have the signature of "generateGenesis" (line 36). The function can then create the DID document and return this Genesis DID document along with ID. 

peerDID, genesisDoc, err := peer.GenerateDID([]did.PublicKey, []did.VerificationMethod)

Framework should define how the Genesis Doc is created as same doc need to be validated at the peer agent (deterministic).

Reply:
@talwinder50 `NewDid` would be a better name, ie. `peer.NewDid()`

Reply:
Done

Reply:
BTW - I would expect a function named “NewDoc” to return some doc struct. Your comment says you are creating a DID?